retrieval/scrape.go

```diff
@@ -496,10 +496,10 @@ func (sl *scrapeLoop) report(start time.Time, duration time.Duration, err error)
 		Value:     model.SampleValue(duration.Seconds()),
 	}
 
-	if err := sl.reportAppender.Append(healthSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), healthSample); err != nil {
 		log.With("sample", healthSample).With("error", err).Warn("Scrape health sample discarded")
 	}
-	if err := sl.reportAppender.Append(durationSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), durationSample); err != nil {
```

Comment:
A bit too much TODO for my taste that's to easy to never address. We should think about this now instead. FWIW, we cannot to a context with timeout here anyway as it would mean partial scrapes.

Reply:
FWIW I really miss this context now I am trying to make sense of OpenTracing spans that aren't connected.